lib/KSelect/index.vue

```diff
@@ -60,123 +167,606 @@
         default: false,
       },
-      clearText: {
-        type: String,
-        default: '',
-      },
     },
+
     data() {
       return {
+        query: '',
+        isInsideModal: false,
+        isActive: false,
+        isTouched: false,
+        highlightedOption: null,
+        showDropdown: false,
+        initialValue: JSON.stringify(this.value),
+        quickMatchString: '',
+        quickMatchTimeout: null,
+        scrollableAncestor: null,
+        dropdownButtonBottom: 'auto',
+        maxDropdownHeight: 256,
         // workaround for Keen-ui not displaying floating labels for empty objects
         selection: Object.keys(this.value || {}).length === 0 ? '' : this.value,
       };
     },
+
     computed: {
       name() {
         return `k-select-${this._uid}`;
       },
+      classes() {
+        return [
+          `ui-select-type-basic`,
+          { 'is-active': this.isActive },
+          { 'is-invalid': this.invalid },
+          { 'is-touched': this.isTouched },
+          { 'is-disabled': this.disabled },
+          { 'has-label': this.hasLabel },
+          { 'has-floating-label': this.hasFloatingLabel },
+          { 'k-select-inline': this.inline },
+          { 'k-select-disabled': this.disabled },
+        ];
+      },
+
+      labelClasses() {
+        return {
+          'is-inline': this.hasFloatingLabel && this.isLabelInline,
+          'is-floating': this.hasFloatingLabel && !this.isLabelInline,
+        };
+      },
+
+      hasLabel() {
+        return Boolean(this.label) || Boolean(this.$slots.default);
+      },
+
+      hasFloatingLabel() {
+        return this.hasLabel && this.floatingLabel;
+      },
+
+      isLabelInline() {
+        return this.selection.length === 0 && !this.isActive;
+      },
+
+      hasFeedback() {
+        return Boolean(this.invalidText) || Boolean(this.$slots.error);
+      },
+
+      showError() {
+        return this.invalid && (Boolean(this.invalidText) || Boolean(this.$slots.error));
+      },
+
+      filteredOptions() {
+        return this.options.filter((option, index) => {
+          return this.defaultFilter(option, index);
+        });
+      },
+
+      displayText() {
+        return this.selection ? this.selection[this.keys.label] || this.selection : '';
+      },
+
+      hasDisplayText() {
+        return Boolean(this.displayText.length);
+      },
+
+      hasNoResults() {
+        if (this.query.length === 0) {
+          return false;
+        }
+
+        return this.filteredOptions.length === 0;
+      },
+
+      submittedValue() {
+        // Assuming that if there is no name, then there's no
+        // need to computed the submittedValue
+        if (!this.name || !this.selection) {
+          return;
+        }
+
+        if (Array.isArray(this.selection)) {
+          return this.selection.map(option => option[this.keys.value] || option).join(',');
+        }
+
+        return this.selection[this.keys.value] || this.selection;
```

Comment:
same thing here, using this.selection and this.value should be consistent! probably  ` return this.value[this.keys.value] || this.value;` would be better. I see more of those cases in this computed property.